Steps to remove sensitive data from repo (#1247)
aguirre-lab/ml4c3: CONTRIBUTING.md

1. [:rocket: Pull requests](#rocket-pull-requests)
1. [:construction: Code reviews](#construction-code-reviews)
1. [:globe_with_meridians: Wiki](#globe_with_meridians-wiki)

> 1. [:scream: I accidentally pushed sensitive data](#scream-i-accidentally-committed-sensitive-info-what-do-i-do)

Reviewer: 🤣 
also the text says pushed, but the linked header says committed; make consistent?

Do not directly edit the `ml4c3` wiki! Instead, submit issues and PRs to the [wiki repo](https://github.com/aguirre-lab/ml4c3-wiki/).
The `ml4c3` wiki is automatically updated from the `ml4c3-wiki` repo via Travis.

## :scream: I accidentally committed sensitive info! What do I do?
Don't worry, there is a fix for that!

> **Note**: This process will affect opened PR and branches, so prevent the rest of the
> team before doing it. Minimize the number of opened branches and PR on the repo by
> merging as many of them as possible.

Reviewer: ```suggestion
**Note**: This process will affect opened PR and branches. You must tell the team to pause work, not submit new PRs, etc. Also, minimize the number of opened branches and PRs by merging as many of them as possible.
```

Sources:
[1](https://stackoverflow.com/questions/48267025/how-to-sync-local-history-after-massive-git-history-rewrite),
[2](https://stackoverflow.com/questions/4084868/how-do-i-recover-resynchronise-after-someone-pushes-a-rebase-or-a-reset-to-a-pub)

> [3](https://docs.github.com/en/github/authenticating-to-github/removing-sensitive-data-from-a-repository)

Reviewer: Move to bottom of section

1. Notify Github support with
   [this form](https://support.github.com/contact/sensitive-data).
   When filling out the form, detail the file and lines where the sensitive data is
   added, as well as a link to the file. Explain that you will run
   `git filter-branch` and ask them to remove cached views and references to the
   sensitive PR. Provide them a link to that PR that introduced the data too.

1. Clean the remote history:
    1. Start from a newly cloned version of the repo:
       ```
       git clone https://github.com/aguirre-lab/ml4c3.git
       cd ml4c3
       ```

>     2. Now you have clean the sensitive file from all the commit history. To do so:

Reviewer: ```suggestion
    2. Remove the sensitive file from the repo's commit history. To do so:
```

      git checkout <branch>
      git reset --hard origin/<branch>
      ```
      Where `<branch>` will probably be master but could also be a recently opened
      branch without changes or with new files still not known by git.

      Then you can copy back the backed-up files.

   1. You have changes committed changed and not pushed them, so you have local commits
   that you want to keep. Option B would reset your branch to its remote version,
   even if you backup the changed files, you will loose the part of the commit history
   that differes from the remote version. If you want to preserve the commits as they
   were on your local branch, follow this workaround:

   Create a copy of your branch before fetching:
   ```
   # BEFORE FETCHING!
   git checkout <branch>
   git branch old-<branch> origin/<branch>
   ```
   This will create a new branch called `old-<branch>` that will keep the contents
   and commit history of your current state of the branch. Now you can fetch and
   then rebase remote branch with your local one. Then you can delete the `old-<branch>`
   ```
   git fetch
   git rebase --onto origin/foo old-foo foo
   git branch -D old-foo
       ```

> 1. At this point, the team can resume work on the repo normally. You can reach bach to
> the GitHub team to notify them that they can perform the rest of the cleaning now.
> Once they finish, the repo will be completely free of sensitive data.

Reviewer: ```suggestion
1. At this point, the team can resume work on the repo normally. Inform GitHub that they can perform the rest of the cleaning. Once they finish, the repo will be completely free of sensitive data.
```
